install_config/sssd_for_ldap_failover.adoc

```diff
@@ -0,0 +1,427 @@
+
+====
+----
+# realm join ldap.example.com
+----
+====
+
+If you are not running a domain, then use the *authconfig* tool:
+
+====
+----
+# authconfig --update --enablesssd --enablesssdauth \
+             --ldapserver=ldap.example.com \
+             --enableldaptls \
+             --ldaploadcert=http://ldap.example.com/ca.crt
+----
+====
+
+This will create *_/etc/sssd/sssd.conf_* with most of the appropriate settings.
+
+[NOTE]
+====
+In RHEL 7, *authconfig* does not create the *_/etc/openldap/cacerts_* directory,
+so you may need to create it manually before running the above command.
+====
+
+If you want to use SSSD to manage failover situations for LDAP, this can be
+configured simply by adding additional entries in *_/etc/sssd/sssd.conf_* on the
+*ldap_uri* line. Systems enrolled with FreeIPA will automatically handle
+failover using DNS SRV records.
+
+By default, SSSD only collects a limited set of attributes about the user from
+the LDAP server (those necessary for local login information). The
+authenticating proxy will want to add the users’ email addresses to its
+output, so you need to configure SSSD to retrieve those as well.
+
+Add the following to the end of the *[domain/domainname]* section of
+*_/etc/sssd/sssd.conf_*:
+
+====
+----
+ldap_user_extra_attrs = mail
+----
+====
+
+Create the *[ifp]* section if it is not present:
+
+====
+----
+[ifp]
+user_attributes = +mail
+----
+====
+
+Restart SSSD to ensure that all of the changes are applied properly:
+
+====
+----
+$ systemctl restart sssd.service
+----
+====
+
+Test that the user information can be retrieved properly:
+
+====
+----
+$ getent passwd <username>
+username:*:12345:12345:Example User:/home/username:/usr/bin/bash
+$ gdbus call \
+        --system \
+        --dest org.freedesktop.sssd.infopipe \
+        --object-path /org/freedesktop/sssd/infopipe/Users/example_2ecom/12345 \
+        --method org.freedesktop.DBus.Properties.Get \
+        "org.freedesktop.sssd.infopipe.Users.User" "extraAttributes"
+(<{'mail': ['username@example.com']}>,)
+----
+====
+
+Attempt to log into the VM as an LDAP user and confirm that the authentication
+is properly set up. This can be done via the local console or a remote service
+such as SSH.
+
+[[phase-2-step-3-apache-configuration]]
+=== Step 3: Apache Configuration
+
+You need to set up Apache to communicate with SSSD. First, create a PAM stack
+file for use with Apache. Create the *_/etc/pam.d/openshift_* file and add the
+following contents:
+
+====
+----
+auth required pam_sss.so
+account required pam_sss.so
+----
+====
+
+This will tell PAM (the pluggable authentication module) that when an
+authentication request is issued for the *openshift* stack, it should use
+*pam_sss.so* to determine authentication and access control.
+
+Next, configure the Apache *_httpd.conf_*. The steps in this section focus on
+setting up the challenge authentication, which is useful for logging in with `oc
+login` and similar automated tools.
+
```

Comment:
There are details in https://sgallagh.wordpress.com/2016/05/27/openshift-and-sssd-part-2-ldap-form-authentication/ that describe whence to get a sample/template login screen that can be dropped in place at this point in the document.